File: docs/sdk-documentation/getting-started/import-and-initialize/swift/customEventLogger.swift
// Example implementation
public class CustomEventLogger : ContextEventLogger {
    public func handleEvent(context: Context, event: ContextEventLoggerEvent) {
        switch event {
        case let .exposure(exposure):
            print("exposed to experiment: \(exposure.name)")
        case let .goal(goal):
            print("goal tracked: \(goal.name)")
        case let .error(error):
            print("error: ", error.localizedDescription)
        case let .publish(event):
            break
        case let .ready(data):
            break
        case let .refresh(data):
            break
        case .close:
            break
        }
    }
}

// for all contexts, during sdk initialization
let absmartlyConfig = ABSmartlyConfig()
absmartlyConfig.contextEventLogger = CustomEventLogger()

// OR, alternatively, during a particular context initialization
let contextConfig = ContextConfig()
contextConfig.eventLogger = CustomEventLogger()

File: docs/SDK Documentation/getting-started/custom-event-logger/dotnet/customEventLogger.cs
    // example implementation
    public class CustomEventLogger : IContextEventLogger
    {
        public void HandleEvent(Context context, EventType eventType, object data)
        {
            switch (eventType)
            {
                case EventType.Exposure when data is Exposure exposure:
                    Console.WriteLine($"exposed to experiment: {exposure.Name}");
                    break;
                case EventType.Goal when data is GoalAchievement goal:
                    Console.WriteLine($"goal tracked: {goal.Name}");
                    break;
                case EventType.Error:
                    Console.WriteLine($"error: {data}");
                    break;
                case EventType.Close:
                case EventType.Publish:
                case EventType.Ready:
                case EventType.Refresh:
                    break;
            }
        }
    }
